basic.C

```diff
@@ -358,7 +358,6 @@ GLOBALFUN string StringHex(UINT32 l, UINT32 digits) {
   digits = 31 - digits;
   while (i > digits) {
     i--;
-    buffer[i] = '0';
```

Comment:
with this change the parameter 'digits' becomes useless.
I would prefer if we do not lose the functionality of the padding zeros
Maybe it would be clearer to construct the mirror image of the padded hex string first
and then flip it.

Reply:
I've opened an issue. I'll drop this patch for now.